functionconvergence.go

```diff
@@ -0,0 +1,51 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize
+
+import "math"
+
+// FunctionConvergence tests for the convergence of function values. See comment
+// in Settings.
+type FunctionConvergence struct {
+	Absolute   float64
+	Relative   float64
+	Iterations int
+
+	absBest float64
+	relBest float64
+	absIter int
+	relIter int
+}
+
+func (fc *FunctionConvergence) Init(f float64) {
+	fc.absBest = f
+	fc.relBest = f
+	fc.absIter = 0
+	fc.relIter = 0
+}
+
+func (fc *FunctionConvergence) FunctionConverged(f float64) Status {
```

Comment:
It's harder to meet than either individually, but in most cases the net effect shouldn't be that large. If one is "much more important" than the contribution of the other is negligible, with the limiting case that the behavior is the same if one or the other is zero. The largest difference in behavior is when they're the similar order of magnitude, which is a factor of 2. This doesn't that different. Thus, the user can still think of the convergence parameters individually when choosing values, and if they really care this rule is much clearer than the currently implemented rule. 

So, SGTM.